common/src/test/java/io/druid/collections/StupidPoolTest.java

```diff
@@ -72,10 +70,24 @@ public void testExceptionInResourceHolderGet() throws IOException
     resourceHolderObj.get();
   }
 
-  @Test
-  public void testFinalizeInResourceHolder()
+  @Test(timeout = 60_000)
+  public void testResourceHandlerClearedByJVM()
+  {
+    if (System.getProperty("java.version").startsWith("1.7")) {
+      // This test is unreliable on Java 7, probably GC is not triggered by System.gc()
+      return;
+    }
```

Comment:
why is this any more reliable in java versions lower or higher, even in java 1.8, System.gc() is just a hint.